drivers/linux/eth/ionic/ionic_bus_pci.c

```diff
@@ -274,9 +275,18 @@ static int ionic_sriov_configure(struct pci_dev *pdev, int num_vfs)
 
 static void ionic_clear_pci(struct ionic *ionic)
 {
-	ionic_unmap_bars(ionic);
-	pci_release_regions(ionic->pdev);
-	pci_disable_device(ionic->pdev);
+	if (ionic->num_bars) {
+		ionic->idev.dev_info_regs = NULL;
+		ionic->idev.dev_cmd_regs = NULL;
+		ionic->idev.intr_status = NULL;
+		ionic->idev.intr_ctrl = NULL;
+
+		ionic_unmap_bars(ionic);
+		pci_release_regions(ionic->pdev);
```

Comment:
Should we wait to update this until the pci_release_regions() doesn't depend on ionic->num_bars?

Reply:
> Should we wait to update this until the pci_release_regions() doesn't depend on ionic->num_bars?

This is what matches the ionic 23.12.2-001 code, whether or not it is optimal.  The purpose here in dsc-drivers is to make the current oot code available, so I'd prefer to keep with this for now.